### ISSUE TITLE
Virtual Machine Properties not shown

### ISSUE THREAD

**Issue by DanteNahuel** [Azure/azure-powershell#11296]
## Description

Some Azure Virtual Machine properties that are shown in the portal, for example 
Operating system: Windows Server Datacenter 2016

Are not shown using the Azure Powershell cmdline.
## Steps to reproduce

```powershell

$vm = get-azvm -name "vmname" -resourcegroupname "rg"
$os = $vm.Storageprofile.Imagereference
```

As mentioned, this comes empty, but the Portal is showing the exact Windows OS Version

More information:
Using a REST API GET call to the InstanceView of the VM brings a JSON body with the OS Version.
`GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/instanceView?api-version=2019-07-01`

Calling the instanceview of the VM using Powershell comes null

`$vm.InstanceView -eq $null
True`

I'm using lastest Az Module.
This happens both in Powershell 7 and in Powershell 5.1

**Comment by DanteNahuel** [comment]
More information:
Using a REST API GET call to the InstanceView of the VM brings a JSON body with the OS Version.
`GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/instanceView?api-version=2019-07-01
`
Calling the instanceview of the VM using Powershell comes null

`$vm.InstanceView -eq $null True`

**Comment by DanteNahuel** [comment]
It seems this call:

`Get-AzVm -Name $vm -ResourceGroupName $rg -Status`

Calls the Instance View API

But why this one doesn't, and comes $null I don't know:

`$vm.InstanceView`

This causes a very bad user experience because there's no explanation on why the InstanceView property, which one would assume is the correct one for calling the API that has the same name, returns null. 
No error, no log, nothing. The reply is OK, but null.

**Comment by dingmeng-xue** [comment]
Get-AzVm only calls below URL and no further call to fetch instance view.
https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-07-01

Feature request to this cmdlet.

**Comment by DanteNahuel** [comment]
> Get-AzVm only calls below URL and no further call to fetch instance view.
> [https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}?api-version=2019-07-01](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/resourceGroups/%7BresourceGroupName%7D/providers/Microsoft.Compute/virtualMachines/%7BvmName%7D?api-version=2019-07-01)
> 
> Feature request to this cmdlet.

I agree
but the question is
Why the -status flag does call the Instance View does while the instanceview property doesnt?

**Comment by Drewm3** [comment]
Please see my comment for issue https://github.com/Azure/azure-rest-api-specs/issues/8678.

I believe the difference may be between the OS information coming from the image details vs. the information coming from the agent. The image details is part of the VM model while the runtime state from the agent is part of the instance view. Generally, the model should not change over time unless there is a control plane operation while the instance view can change due to changes in the guest operating system itself without any change to the control plane.

Let me know if this answers the question.

**Comment by DanteNahuel** [comment]
> Please see my comment for issue [Azure/azure-rest-api-specs#8678](https://github.com/Azure/azure-rest-api-specs/issues/8678).
> 
> I believe the difference may be between the OS information coming from the image details vs. the information coming from the agent. The image details is part of the VM model while the runtime state from the agent is part of the instance view. Generally, the model should not change over time unless there is a control plane operation while the instance view can change due to changes in the guest operating system itself without any change to the control plane.
> 
> Let me know if this answers the question.

It does, but now I know there's a coherence issue between Powershell properties and cmdlets.
I'll close this issue and open a new one, more specific.
Thanks @Drewm3

**Comment by expoearchie** [comment]
Having had to compile a list of Azure VMs and Operating Systems using Powershell recently I had exactly this issue. To resolve, this is what I did:

If you run

$vm = Get-AzVM -Name xxxx -ResourceGroup yyyy
$os = $vm.Storageprofile.ImageReference.Sku


The variable $os will retrieve the Image SKU details if the VM was deployed from the Azure marketplace. In most cases, this will suffice, but if the customer's environment is using a lot of older VMs and/or migrated VMs, you can retrieve the data reported by the Azure extension on the VM by using the following powershell:

$vmInstanceView = Get-AzVM -Name xxxx -ResourceGroup yyyy -Status
$osName = $vminstanceview.OsName

Basically the Instance view details of the VM is achieved by using the Get-AzVM -Status flag.
